pkg/odo/util/validation.go

```diff
@@ -22,3 +24,82 @@ func ValidateName(name string) error {
 	return nil
 
 }
+
+func validateNameAsValidator(name interface{}) error {
+	s := name.(string)
+	return ValidateName(s)
+}
+
+const (
+	// NameValidatorKey is the key used to identify the validator associated with ValidateName function
+	NameValidatorKey           = "odo_name_validator"
+	defaultIntegerValidatorKey = "odo_default_integer"
+)
+
+// Validatable represents a common ancestor for validatable parameters
+type Validatable struct {
+	Required             bool
+	Type                 string
+	AdditionalValidators []string
+}
+
+// AsValidatable allows avoiding type casts in client code
+func (v Validatable) AsValidatable() Validatable {
+	return v
+}
+
+// Validator is a function that validates that the provided interface is conform to expectations or return an error
```

Comment:
` that the provided interface is conform to` -> ` that the provided interface conforms to`

Reply:
Fixed. Thanks.